.github/workflows/ipfs.yml

```diff
@@ -0,0 +1,28 @@
+---
+name: Deploy affected projects to IPFS
+
+'on':
+  # Triggers the workflow on push to main branch
+  push:
+    branches: [master]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch: null
+
+jobs:
+  master:
+    name: Publish NX Affected
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          fetch-depth: 0
+      - name: Use Node.js 16
+        id: Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.14.0
+      - name: Run deploy for affected sites
+        run: yarn nx print-affected --base=origin/master~1 --head=origin/master --select=projects | ./tools/ipfs-deploy.js
```

Comment:
different between origin/master~1 and origin/master is always latest commit. If we will push 5 new commit to master and only changes from top commit will be taken to comparsion.
We should do --base=REF_TO_COMMIT_PREVIOUSLY_DEPLOYED --head=REF_TO_COMMIT_WHICH_WE_WANT_DEPLOY

We need to take information some somewhere which version is currently deployed (I will suggest git tag).
We can have tag for currently deployed version to IPSF and update it after each deployment
```
git checkout master
git pull
yarn nx print-affected --base=IPFS --head=master --select=projects | ./tools/ipfs-deploy.js
git tag -a IPFS -m "Some extra comment, maybe even date"
git push origin IPFS
```

Reply:
It was a fair point. I've rewritten the script so that instead of this workflow, or the one it was before, it:

1. Checks the ./apps/** folder to see which one has a fleek config
2. Gets the fleek siteId for each project
3. Uses their API to fetch the last commit that triggered a build
4. Runs nx affected and checks to see if this project was affected
5. If it was, trigger a build using their API

It feels circular and weird, but when I started looking at writing an nx command that would return it's siteID, or better yet, the last commit ID, it didn't seem like there would be a good way to filter nx affected... so I bailed out and used this shell script style instead. I might be fighting the tools though.